rfcs/0237-vm-syscalls-2/0237-vm-syscalls-2.md

```diff
@@ -0,0 +1,87 @@
+---
+Number: "0237"
+Category: Standards Track
+Status: Proposal
+Author: Wanbiao Ye
+Organization: Nervos Foundation
+Created: 2021-05-25
+---
+
+# VM Syscalls 2
+
+## Abstract
+
+This document describes the addition of the syscalls during the ckb's first hard fork.
+
+- [VM Version]
+- [Current Cycles]
+- [Exec]
+
+### VM Version
+[vm version]: #vm-version
+
+As shown above, *VM Version* syscall has a signature like following:
+
+```c
+int ckb_vm_version()
+{
+  return syscall(2041, 0, 0, 0, 0, 0, 0);
+}
+```
+
+*VM version* syscall returns current running VM version, so far 2 values will be returned: 0 for Lina CKB-VM version, 1 for the new hardfork version. This syscall consumes 500 cycles.
+
+### Current Cycles
+[current cycles]: #current-cycles
+
+*Current Cycles* syscall has a signature like following:
+
+```c
+uint64_t ckb_current_cycles()
+{
+  return syscall(2042, 0, 0, 0, 0, 0, 0);
+}
+```
+
+*Current Cycles* returns current cycle consumption just before executing this syscall. This syscall consumes 500 cycles.
+
+
+### Exec
+[exec]: #exec
+
+Exec runs an executable file from specified cell data in the context of an already existing machine, replacing the previous executable. The used cycles does not change, but the code, registers and memory of the vm are replaced by those of the new program. It's cycles consumption consists of two parts:
+
```

Comment:
Will the code, register and memory be restored to the previous executable program, after the specified new executable file execution finished?

Reply:
No